source/options.tsx

```diff
@@ -99,8 +153,13 @@ async function generateDom(): Promise<void> {
 
 function addEventListeners(): void {
 	// Update domain-dependent page content when the domain is changed
-	select('.js-options-sync-selector')?.addEventListener('change', ({currentTarget: dropdown}) => {
-		select<HTMLAnchorElement>('#personal-token-link')!.host = (dropdown as HTMLSelectElement).value;
+	select('.OptionsSyncPerDomain-picker select')?.addEventListener('change', ({currentTarget: dropdown}) => {
+		const host = (dropdown as HTMLSelectElement).value === 'default' ? 'github.com' : (dropdown as HTMLSelectElement).value;
+		select<HTMLAnchorElement>('#personal-token-link')!.host = host;
+		// Delay validating to let options load first
+		setTimeout(() => {
+			void validateToken();
+		}, 100);
```

Comment:
```suggestion
		setTimeout(validateToken, 100);
```

But better yet if you dispatch an event at the end of https://github.com/fregante/webext-options-sync-per-domain/blob/dcc61fe58ab67d0982b2948cf3c1264375ef4096/index.ts#L129

Example: https://github.com/fregante/webext-options-sync/blob/master/index.ts#L253

… and then listen to that event specifically, instead of changing this listener

Reply:
I tried to figure this out. But I could not get it to work by listening to the event.